android-client-sdk/src/main/java/com/devcycle/sdk/android/model/Variable.kt

```diff
@@ -93,9 +81,9 @@ class Variable<T> internal constructor(
 
     @Throws(IllegalArgumentException::class)
     @Suppress("UNCHECKED_CAST")
-    private fun updateVariable(variable: Variable<Any>) {
+    private fun updateVariable(variable: ReadOnlyVariable<Any>) {
         var executeCallBack = false
-        if (variable.type != type) {
+        if (getType(variable.value) != type) {
```

Comment:
is the `type` field on the read only variable not a reliable field? Why do we need to derive it from the value? I noticed its also being derived in the `initializeFromVariable` function below

Reply:
It just doesn't seem necessary to trust that the server is sending that type correctly when we can see from the type of the "value" field what the type actually is. I've been making the same change across all the SDKs.

Reply:
Yeah I agree with that. Is it maybe worth getting rid of the field altogether then rather than risk a situation where the type field is wrong and confuses users? Or is that just a can of worms

Reply:
I guess that just fucks our whole data model up so its kinda pointless

Reply:
i thought about it, but it seemed too much of a departure from what we do elsewhere. Feels like something we need to decide to do across all the SDKs

Reply:
👍